Content.Server/Chat/LastMessageBeforeDeath.cs

```diff
@@ -0,0 +1,199 @@
+                if (playerMessages != null)
+                {
+                    if (playerMessages["EntityUid"] is EntityUid playerUid && _mobStateSystem.IsDead(playerUid))
+                    {
+                        foreach (DictionaryEntry playerName in playerMessages)
+                        {
+                            if (playerName.Key.ToString() != "EntityUid")
+                            {
+                                allMessagesString.AppendLine($"{playerName.Value}");
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (allMessagesString.ToString().Length == 0)
+            {
+                allMessagesString.AppendLine($"No messages found.");
+            }
+
+            var messages = SplitMessage(allMessagesString.ToString(), MaxMessageSize);
+            var discordWebhook = new DiscordWebhook();
+            int messageCount = 0;
+            foreach (var message in messages)
+            {
+                if (messageCount >= MaxMessagesPerBatch)
+                {
+                    await Task.Delay(60000); // Wait for 1 minute
+                    messageCount = 0;
+                }
+                var payload = new WebhookPayload { Content = message };
+                var response = await discordWebhook.CreateMessage(id, payload);
+                messageCount++;
+
+                // Insert a small delay between each message
+                await Task.Delay(MessageDelayMilliseconds);
+
+                // Response still can be handled if needed.
+#if DEBUG
+                if (response.IsSuccessStatusCode)
+                {
+                    _sawmill.Debug("Last Messages Before Death sent successfully.");
+                }
+                else
+                {
+                    _sawmill.Error($"Failed to send last messages before death. Status code: {response.StatusCode}");
+                }
+#endif
+            }
+        }
+
+
+
+        private List<string> SplitMessage(string message, int chunkSize)
+        {
+            var messages = new List<string>();
+            int start = 0;
+
+            while (start < message.Length)
+            {
+                int end = start + chunkSize;
+
+                if (end >= message.Length)
+                {
+                    messages.Add(message.Substring(start));
+                    break;
+                }
+
+                int lastNewLine = message.LastIndexOf('\n', end);
+                if (lastNewLine > start)
+                {
+                    messages.Add(message.Substring(start, lastNewLine - start));
+                    start = lastNewLine + 1;
+                }
+                else
+                {
+                    messages.Add(message.Substring(start, chunkSize));
+                    start += chunkSize;
+                }
+            }
+
+            return messages;
+        }
+
```

Comment:
This should be on the discord manager I mentioned rather than being a singleton tucked away in here.